### PR TITLE
Update python scripts

### PR DESCRIPTION
**Pull request by maxGimeno**
## Summary of Changes

Update the python scripts in Documentation so they work with python3. 
## Release Management

* Affected package(s):Documentation

### CODE REVIEW

File: Documentation/doc/scripts/test_doxygen_versions.sh [hunk 1]
```diff
@@ -126,6 +128,7 @@ if [ "$HAS_REF" -ne "1" ]; then
     CGAL_NAME="$(cat $ROOT/VERSION)"
     cd $ROOT/doc/scripts
   fi
+  echo "CGAL_NAME = $CGAL_NAME"
```
Comment (lrineau) [inline Documentation/doc/scripts/test_doxygen_versions.sh]: This line seems a leftover of a debug session.

Comment (maxGimeno) [reply Documentation/doc/scripts/test_doxygen_versions.sh]: Indeed

**Comment by sloriot** [comment]
No doc generated, this PR is probably responsible.

**Comment by maxGimeno** [comment]
> No doc generated, this PR is probably responsible.

I also received a mail from the cron stating:
Traceback (most recent call last):
  File "../documentation_parser.py", line 36, in <module>
    e = pq(file_content, parser="xml")
  File "/home/cgal-testsuite/.local/lib/python3.6/site-packages/pyquery/pyquery.py", line 267, in __init__
    raise TypeError(context)
TypeError: <codecs.StreamReaderWriter object at 0x7feb7920b978>

**Comment by maxGimeno** [comment]
I think we need to update the cron job that manages the doc generation to make it call python3. Without that this PR won't work.

**Comment by lrineau** [comment]
How can we test it "locally"?

**Comment by maxGimeno** [comment]
I don't know what the cronjob does exactly, so I can't be sure, but for the develepment I used [this branch](https://github.com/CGAL/cgal-documentation-dockerfile/pull/1) to generate the doc using the python3 scripts.

**Comment by lrineau** [comment]
The crontab of cgal-testsuite@cgal.geometryfactory.com is:
```
PATH=/home/cgal-testsuite/bin:/usr/local/bin:/bin:/usr/bin
LC_CTYPE=en_US.UTF-8

# run doxygen testsuite at 21:23
23 21  *  *  *   source /opt/rh/devtoolset-8/enable && /home/lrineau/bin/run_doxygen_testsuite
```

And `~cgal-testsuite/bin/` contains two symbolic links:
```
cmake -> /usr/bin/cmake3
flex -> /home/cgal-testsuite/Downloads/flex-2.6.4/build/install/bin/flex
```
to use CMake version 3.14.7 and flex version 2.6.4.

`/home/lrineau/bin/run_doxygen_testsuite` is a symbolic link to `/home/lrineau/latest-internal-release-branch/Scripts/developer_scripts/run_doxygen_testsuite`.

**Comment by lrineau** [comment]
But I think we should reconsider using a Docker container to build the documentation.

**Comment by maxGimeno** [comment]
Why don't we use the docker container in the branch I pointed ?

**Comment by lrineau** [comment]
Yes, of course. That is the reason why we made you work on that Docker image. I think we should test this image, with the Python3 branch, and eventually switch to using that Docker image officially.

**Comment by sloriot** [comment]
Why is this branch not based on 4.14? Why is it not integrated? Anyone without python2 on his system will not be able to build the doc.

**Comment by lrineau** [comment]
The problem with that branch is that it removes the support for Python2. We cannot merge it blindly as it is.

**Comment by sloriot** [comment]
OK then we can extract my last commit in a separate PR as it fixes the issue for @MaelRL . Would it be OK?

**Comment by lrineau** [comment]
> OK then we can extract my last commit in a separate PR as it fixes the issue for @MaelRL . Would it be OK?

Yes, that is the best solution, for the moment.

**Comment by sloriot** [comment]
Done in #4858, I removed the commit from this branch.

**Comment by sloriot** [comment]
@maxGimeno  We have to retest it, could you please refresh it?

**Comment by maxGimeno** [comment]
```
 * branch            HEAD       -> FETCH_HEAD
Traceback (most recent call last):
  File "../documentation_parser.py", line 38, in <module>
    e = pq(file_content, parser="xml")
  File "/home/cgal-testsuite/.local/lib/python3.6/site-packages/pyquery/pyquery.py", line 267, in __init__
    raise TypeError(context)
TypeError: <codecs.StreamReaderWriter object at 0x7fe97a657128>
```

**Comment by maxGimeno** [comment]
Successfully tested in https://cgal.geometryfactory.com/CGAL/testsuite/results-5.4-Ic-42.shtml

File: Documentation/doc/scripts/process_doc.sh [hunk 2]
```diff
@@ -52,13 +52,17 @@ cd $PATH_TO_SCRIPTS #scripts
 PATH_TO_MASTER="$PWD/doxygen_master/build/bin/doxygen"
 echo "done."
 
-echo "comparing versions 1.8.4 and 1.8.13"
-bash -$- test_doxygen_versions.sh $PATH_TO_1_8_4 $PATH_TO_1_8_13 $PWD/doc_1_8_4 $PWD/doc_1_8_13 $PUBLISH_DIR
+echo "comparing versions 1.8.18 and 1.8.13"
+bash -$- test_doxygen_versions.sh $PATH_TO_1_8_13 $PATH_TO_1_8_18 $PWD/doc_1_8_13 $PWD/doc_1_8_18  $PUBLISH_DIR
+if [ ! -d $PWD/doc_1_8_18/doc_log ]; then
```
Comment (sloriot) [inline Documentation/doc/scripts/process_doc.sh:57]: does it mean that `doc_1_8_13/doc_log` always if it exists for 1.8.18 ?

File: Documentation/doc/scripts/test_doxygen_versions.sh [hunk 3]
```diff
@@ -94,7 +94,7 @@ echo "done."
 #add post-processing
 cd ./build_doc
 echo "Adding postprocessing..."
-make -j$NB_CORES doc_with_postprocessing  &>> ./build_logs
+make -j$NB_CORES doc_with_postprocessing  & >> ./build_logs
```
Comment (sloriot) [inline Documentation/doc/scripts/test_doxygen_versions.sh:97]: @lrineau this change is not correct, right?

File: Documentation/doc/scripts/testsuite.py [hunk 4]
```diff
@@ -74,7 +74,7 @@ def write_report(args):
 </head><body>
 <h1 id="maintitle">Doxygen Manual Results</h1>'''
     page_footer='''<table border="1" cellspacing="2" cellpadding="5" class="test-results">
-    <tr><td/><th colspan="3">Doxygen 1.8.4</th><th colspan="3">Doxygen 1.8.13(official)</th><th colspan="3">Doxygen master</th></tr>
+    <tr><td/><th colspan="3">Doxygen 1.8.13(official)</th><th colspan="3">Doxygen 1.8.18</th><th colspan="3">Doxygen master</th></tr>
```
Comment (sloriot) [inline Documentation/doc/scripts/testsuite.py:77]: official is the second one then, right?

**Review by sloriot — COMMENTED** [review]
@lrineau `process_doc.sh` and `testsuite.py` were not tested but I looked at them and added some comments where I have a doubt.

File: Documentation/doc/scripts/process_doc.sh (hunk 2 above)
Comment (maxGimeno) [reply Documentation/doc/scripts/process_doc.sh:57]: yes

File: Documentation/doc/scripts/test_doxygen_versions.sh (hunk 3 above)
Comment (lrineau) [reply Documentation/doc/scripts/test_doxygen_versions.sh:97]: Indeed. The redirection `&>> ./build_logs` is equivalent to `>> ./build_logs 2>&1` (cerr and cout both redirected to the file).

`make -j$NB_CORES doc_with_postprocessing  & >> ./build_logs` launches `make` in the background (without any redirection), and `>> ./build_logs` is a redirection for a non existent command. (Maybe you do not know, but we can write the redirections *before* the command: `> log echo Hello` will write "Hello" in the file `log`.).

File: Documentation/doc/scripts/testsuite.py (hunk 4 above)
Comment (lrineau) [reply Documentation/doc/scripts/testsuite.py:77]: I am surprised: this PR is about the transition from Python2 to Python3, but it seems that, at the same same we have something about the transition from Doxygen 1.8.13 to Doxygen 1.8.18... (and why version 1.8.18 by the way?!).

**Comment by lrineau** [comment]
@sloriot How can we move forward with this PR? Is there a way we can test it?

**Comment by sloriot** [comment]
`PYTHONPATH=/home/cgal-testsuite/.local/lib/python2.6/site-packages` should be updated in `Scripts/developer_scripts/run_doxygen_testsuite` as well as in `/home/lrineau/latest-internal-release-branch/Scripts/developer_scripts/` when we test that branch.

**Comment by sloriot** [comment]
Successfully tested in CGAL-5.4-Ic-57

**Pull request merged**